packages/flutter/test/material/dropdown_test.dart

```diff
@@ -3122,4 +3122,162 @@ void main() {
+                ),
+                DropdownMenuItem<String>(
+                  value: 'second',
+                  child: Text('second'),
+                ),
+              ],
+              onChanged: (String? newValue) {
+                setState(() {
+                  value = newValue;
+                });
+              },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Open dropdown.
+    await tester.tap(find.text('first').hitTestable());
+    await tester.pumpAndSettle();
+
+    // Tap on a disabled item.
+    await tester.tap(find.text('disabled').hitTestable());
+    await tester.pumpAndSettle();
+
+    // The dropdown should still be open, i.e., there should be one widget with 'second' text.
+    expect(find.text('second').hitTestable(), findsOneWidget);
+  });
+
+  testWidgets('Disabled item should not be focusable', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: DropdownButton<String>(
+            value: 'enabled',
+            items: const <DropdownMenuItem<String>>[
+              DropdownMenuItem<String>(
+                enabled: false,
+                child: Text('disabled'),
+              ),
+              DropdownMenuItem<String>(
+                value: 'enabled',
+                child: Text('enabled'),
+              )
+            ],
+            onChanged: onChanged,
+          ),
+        ),
+      ),
+    );
+
+    // Open dropdown.
+    await tester.tap(find.text('enabled').hitTestable());
+    await tester.pumpAndSettle();
+
+    try {
+      // try to get the focus node for disabled item
+      Focus.of(tester.element(find.text('disabled').hitTestable()));
+      // if it is possible to get it, means this disabled item is focusable, which it shouldn't be.
+      fail('Disabled menu item should not be able to request focus');
+    } on AssertionError catch (error) {
+      expect(
+        error.toString(),
+        contains('Focus.of() was called with a context that does not contain a '
+            'Focus between the given context and the nearest FocusScope widget'),
+      );
+    }
+  });
+
+  testWidgets('value should be null if enabled is false', (WidgetTester tester) async {
+    try {
+      // Try to assign [value] and [onTap] non-null values irrespective of
+      // if the menu item is disabled for now
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: DropdownButton<String>(
+              value: 'enabled',
+              items: <DropdownMenuItem<String>>[
+                DropdownMenuItem<String>(
+                  value: 'disabled',
+                  enabled: false,
+                  child: const Text('disabled'),
+                ),
+                const DropdownMenuItem<String>(
+                  value: 'enabled',
+                  child: Text('enabled'),
+                )
+              ],
+              onChanged: onChanged,
```

Comment:
nit: prefer to avoid trailing params since they might be missed
```suggestion
              value: 'enabled',
              onChanged: onChanged,
              items: <DropdownMenuItem<String>>[
                DropdownMenuItem<String>(
                  value: 'disabled',
                  enabled: false,
                  child: const Text('disabled'),
                ),
                const DropdownMenuItem<String>(
                  value: 'enabled',
                  child: Text('enabled'),
                )
              ],
```